prov/rxm/src/rxm_ep.c

```diff
@@ -633,6 +635,132 @@ static ssize_t rxm_rma_iov_init(struct rxm_ep *rxm_ep, void *buf,
 	return sizeof(*rma_iov) + sizeof(*rma_iov->iov) * count;
 }
 
+static inline ssize_t
+rxm_ep_format_transmit_lightweight(struct rxm_ep *rxm_ep, struct rxm_conn *rxm_conn,
+				   size_t len, uint64_t data, uint64_t flags,
+				   uint64_t tag, uint8_t op, struct rxm_tx_buf **tx_buf)
+{
+	*tx_buf = RXM_TX_BUF_GET(rxm_ep);
+	if (OFI_UNLIKELY(!*tx_buf)) {
+		FI_WARN(&rxm_prov, FI_LOG_EP_DATA, "TX queue full!\n");
+		return -FI_EAGAIN;
+	}
+
+	(*tx_buf)->hdr.msg_ep = rxm_conn->msg_ep;
+
+	(*tx_buf)->pkt.ctrl_hdr.version = OFI_CTRL_VERSION;
+	(*tx_buf)->pkt.ctrl_hdr.conn_id = rxm_conn->handle.remote_key;
+	(*tx_buf)->pkt.ctrl_hdr.type = ofi_ctrl_data;
+	(*tx_buf)->pkt.hdr.version = OFI_OP_VERSION;
+	(*tx_buf)->pkt.hdr.op = op;
+	(*tx_buf)->pkt.hdr.size = len;
+	(*tx_buf)->pkt.hdr.tag = tag;
+	(*tx_buf)->pkt.hdr.flags = 0;
+	rxm_op_hdr_process_flags(&(*tx_buf)->pkt.hdr, flags, data);
+
+	return FI_SUCCESS;
+}
+
+static inline ssize_t
+rxm_ep_format_transmit(struct rxm_ep *rxm_ep, struct rxm_conn *rxm_conn,
+		       void *context, uint8_t count, size_t len,
+		       uint64_t data, uint64_t flags, uint64_t tag,
+		       uint8_t op, uint64_t comp_flags,
+		       struct rxm_tx_buf **tx_buf, struct rxm_tx_entry **tx_entry)
+{
+	ssize_t ret;
+
+	ret = rxm_ep_format_transmit_lightweight(rxm_ep, rxm_conn, len, data,
```

Comment:
nit: how about `rxm_ep_format_tx_buf`?

Reply:
I think `rxm_ep_format_tx_res()` will reflect the purpose of this more clearly

Reply:
What does `res` stand for?

Reply:
> What does res stand for?

`res` stands for resources. If needed, I can rename it in the future